deployment/tfhub/object-detection-url.ipynb

```diff
@@ -0,0 +1,625 @@
+        "    def predict(self, data):\n",
+        "        result = []\n",
+        "\n",
+        "        for item in data:\n",
+        "            file_name, url = item\n",
+        "            result.append(self.detect_objects(file_name, url))\n",
+        "\n",
+        "        return result\n",
+        "\n",
+        "    def example(self):\n",
+        "        return [['000001_1.jpg', 'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg']]\n",
+        "\n",
+        "    def describe(self):\n",
+        "        return {\n",
+        "            'method': 'predict',\n",
+        "            'args': f\"{self.example()}\",\n",
+        "            'returns': \"\"\"\n",
+        "                [\n",
+        "                    [{\n",
+        "                        'file_name': '',\n",
+        "                        'has_car': False,\n",
+        "                        'score': 0,\n",
+        "                        'bboxes': {\n",
+        "                            'ymin': 0, \n",
+        "                            'xmin': 0, \n",
+        "                            'ymax': 0,\n",
+        "                            'xmax': 0 \n",
+        "                        }\n",
+        "                    }]\n",
+        "                ]\n",
+        "            \"\"\",\n",
+        "            'description': 'Identify whether a given object is present in the image URL.',\n",
+        "            'input_description': 'Image URL.',\n",
+        "            'output_description': \"\"\"\n",
+        "                A JSON object containing the image file name, a boolean value indicating \n",
+        "                if the object was found in the image, its score and bounding box\n",
+        "            \"\"\"\n",
+        "        }"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5SvthV9gd4lk"
+      },
+      "source": [
+        "## 5. Model Register"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "UR7XtPNOHTgO",
+        "outputId": "99b78ecb-a7e8-452a-91dc-8c9ef7ab0faf"
+      },
+      "outputs": [],
+      "source": [
+        "registered_model = client.get_or_create_registered_model(name='Object Detection - TF Hub')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "vFlrCL2EWnJR",
+        "outputId": "976ab8fc-9343-4ff9-b127-7c3153943a7c"
+      },
+      "outputs": [],
+      "source": [
+        "model = registered_model.create_standard_model(\n",
+        "    name = 'url',\n",
+        "    model_cls = DetectObject,\n",
+        "    model_api = ModelAPI(\n",
+        "        [{'file_name': '', 'url': ''}],\n",
```

Comment:
> Note that ModelAPI input argument had to be a dict: [{'file_name': '', 'url': ''}], with ['file_name', 'img'] the API shows 0 and 1 for these values.

Alright, I understand this reasoning. However, as [I mentioned before](https://vertaai.slack.com/archives/C048J9USF1B/p1672767603106639):

> The most important thing for us is that ModelAPI(input, output) uses the _exact_ input and output shapes from the model. If we make any further adjustments, then we are no longer accurately representing the model’s API.

Preferring `[{'file_name': '', 'url': ''}]` instead of `['file_name', 'img']` is alright, but that means we should _also_ change `DetectObject.predict()` and `DetectObject.example()` to use

```python
[
    {
        'file_name': '000001_1.jpg',
        'img': 'http://www.cs.ucf.edu/~aroshan/index_files/Dataset_PitOrlManh/images/000001_1.jpg',
    }
]
```

ModelAPI's input argument _must_ match the model's actual input.

Reply:
Ok, I kept `[{'file_name': '', 'url': ''}]` and changed `example()` and `predict()` methods to accept a dict.
As you said, this way the ModelAPI input matches exactly the Model input :+1: